### PR TITLE
Use Requirements for custom field instead of manual settings

### PR DESCRIPTION
**Pull request by vaevictis**
My previous Linked Ticket App :octocat: PR relied on the settings mechanism and manual intervention to set a `Linked Data` custom text field that would hold the relationship between 2 tickets.

As it happens, **requirements** are now a thing. So the app was not working on Staging. This is now fixed.

I also took the liberty to add some tests and clean-up the code.

### CODE REVIEW

File: .zat [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "zat_latest": "2.9.3",
+  "zat_update_check": "2018-05-11"
+}
```
Comment (fvzzy) [inline .zat]: Newline

File: spec/src/app_spec.js [hunk 2]
```diff
@@ -255,7 +264,7 @@ describe('App', () => {
   // the scope wrangling that BaseappInstance is doing.
   xdescribe('validateFormFields', () => {
```
Comment (fvzzy) [inline spec/src/app_spec.js]: Re-enable test?

File: src/javascript/app.js [hunk 3]
```diff
@@ -597,42 +612,39 @@ var App = {
   tags: function () {
     var tags = []
 
-    // TODO: de-lodash if possible
-    if (!_.isEmpty(this.ticket.tags)) { tags = _.union(tags, this.ticket.tags) }
-
-    if (!_.isEmpty(this.settings.child_tag)) { tags = _.union(tags, [ this.settings.child_tag ]) }
+    if (Array.isArray(this.ticket.tags)) { tags = this.ticket.tags }
```
Comment (fvzzy) [inline src/javascript/app.js]: Nice 👍

File: src/javascript/app.js [hunk 4]
```diff
@@ -597,42 +612,39 @@ var App = {
...
+    let customFieldName = `ticketFields:custom_field_${this.ancestryFieldId}`
+
+    return this.zafClient.get(customFieldName).then(function (ticketFieldsData) {
       if (!ticketFieldsData.hasOwnProperty(customFieldName)) {
         console.error('error: ', this.I18n.t('ancestry_field_missing'))
         return false
       }
       return this.zafClient.invoke(`${customFieldName}.hide`)
-    })
+    }.bind(this))
```
Comment (fvzzy) [inline src/javascript/app.js]: Would an arrow function work to eliminate the need for context binding here?

File: src/javascript/app.js [hunk 5]
```diff
@@ -597,42 +612,39 @@ var App = {
...
-    if (!_.isEmpty(this.ticket.tags)) { tags = _.union(tags, this.ticket.tags) }
-
-    if (!_.isEmpty(this.settings.child_tag)) { tags = _.union(tags, [ this.settings.child_tag ]) }
+    if (Array.isArray(this.ticket.tags)) { tags = this.ticket.tags }
+    if (this.settings.child_tag && this.settings.child_tag.length !== 0) { tags.push(this.settings.child_tag) }
 
-    return tags
+    // A Set is an iterable data structure that contains only unique values.
+    // This line filters out any duplicate tags.
+    return Array.from(new Set(tags))
```
Comment (fvzzy) [inline src/javascript/app.js]: Nice 👍

File: src/javascript/app.js (hunk 4 above)
Comment (vaevictis) [reply src/javascript/app.js]: It does 👍

File: .zat (hunk 1 above)
Comment (vaevictis) [reply .zat]: I reckon I don't need to submit `.zat` actually.

File: spec/src/app_spec.js (hunk 2 above)
Comment (vaevictis) [reply spec/src/app_spec.js]: Tried again. Still fails for an obscure reason.

Comment (vaevictis) [reply spec/src/app_spec.js]: Tried more. Fixed it 🎉

File: lib/javascript/helpers.js [hunk 6]
```diff
@@ -1,21 +1,25 @@
...
+    }
+    return fieldHasContent
+  },
 
-  if (fieldHasContent) {
-    viewField.removeClass('c-txt__input--error')
-  } else {
-    viewField.addClass('c-txt__input--error')
+  convertLineBreaksToHtml: (strToConvert) => {
+    return strToConvert.replace(/\n/g, '<br>')
```
Comment (fvzzy) [inline lib/javascript/helpers.js]: Any chance these line-breaks ever get read in as `\r\n`? In that case something like `str.replace(/(?:\r\n|\r|\n)/g, '<br>')` might be more robust (just stealing from https://stackoverflow.com/a/784547)

**Comment by fvzzy** [comment]
Looks OK to me other than the comments 👍

File: lib/javascript/helpers.js (hunk 6 above)
Comment (vaevictis) [reply lib/javascript/helpers.js]: I like it.

**Comment by hiroiguchi** [comment]
👌 for string but if you could cc @zendesk/i18n  and split string PR and Code one from next time, that will be much appreciated.

**Pull request merged**